networkx/generators/directed.py

```diff
@@ -241,13 +241,14 @@ def scale_free_graph(n, alpha=0.41, beta=0.54, gamma=0.05, delta_in=0.2,
            Discrete Algorithms, 132--139, 2003.
     """
 
+    # TODO
     def _choose_node(G,distribution,delta):
         cumsum=0.0
         # normalization 
-        psum=float(sum(distribution.values()))+float(delta)*len(distribution)
-        r=random.random()
-        for i in range(0,len(distribution)):
-            cumsum+=(distribution[i]+delta)/psum
+        psum = float(sum(distribution.values())) + float(delta) * len(G)
+        r = random.random()
+        for i in range(0, len(G)):
+            cumsum += (distribution[i] + delta) / psum
```

Comment:
The only way I can think of is to move the calculation of psum outside the function. 
It looks like it is a count of the edges added so far plus delta times the number of nodes added so far.

Maybe you can count the nodes and edges as you add them. Then before calling _choose_node
you can compute ```psum = numb_edges + delta * len(G)```.  The numb_edges is just set to ```0```
before the loop and incremented when the edge is added.

Then lines 250-252 become:

    for n,d in distribution:
        cumsum += d + delta
        if r*psum < cumsum:
            break

Note that this doesn't march through the nodes ```1,2,3,4,..,n``` as before.  Instead you get nodes in the order they are returned by degree().  Check me, but I think that won't matter -- the nodes still get chosen based on
the correct probability, even if they don't get cycled through in the same order.